### PR TITLE
Remove schedule output from ZStream.schedule[Elements]

### LINKED ISSUE #1534: Modify ZStream#schedule and ZStream#scheduleElements to discard the schedule's output

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Closes #1534

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -1671,8 +1671,8 @@ class ZStream[-R, +E, +A](val process: ZManaged[R, E, Pull[R, E, A]]) extends Se
    * Repeats are done in addition to the first execution, so that `scheduleElements(Schedule.once)` means "emit element
    * and if not short circuited, repeat element once".
    */
-  final def scheduleElements[R1 <: R, A1 >: A](schedule: ZSchedule[R1, A, A1]): ZStream[R1 with Clock, E, A1] =
-    scheduleElementsEither(schedule).map(_.merge)
+  final def scheduleElements[R1 <: R, A1 >: A](schedule: ZSchedule[R1, A, _]): ZStream[R1 with Clock, E, A1] =
```
Comment (jdegoes) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]: I'd use `Any` instead of `_`, as existentials are going away in Dotty.

**Comment by iravid** [comment]
Yes! Awesome!

**Comment by iravid** [comment]
Ah tests need to be adjusted

File: streams/shared/src/main/scala/zio/stream/ZStream.scala (hunk 1 above)
Comment (vasilmkd) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: Hmm, let's open an issue that checks for that. I have seen existentials in other places as well.

File: streams-tests/jvm/src/test/scala/zio/stream/StreamSpec.scala [hunk 2]
```diff
@@ -1628,27 +1628,27 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRu
   private def repeatedAndSpaced =
```
Comment (vasilmkd) [inline streams-tests/jvm/src/test/scala/zio/stream/StreamSpec.scala]: Is there a possibility that this test is flaky? Can you take another look @regiskuckaertz ?

Comment (vasilmkd) [reply streams-tests/jvm/src/test/scala/zio/stream/StreamSpec.scala]: Nevermind, bad rebase. Sorry.

**Pull request merged**